app/src/main/java/com/kms/cura/view/activity/SearchActivity.java

```diff
@@ -155,11 +156,31 @@ public boolean onCreateOptionsMenu(Menu menu) {
     private List<DoctorUserEntity> initArray() {
         Gson gson = new Gson();
         doctors = gson.fromJson(data, DoctorUserEntity.getDoctorEntityListType());
+        calculateRatings();
         Collections.sort(doctors, new NameComparator());
         filteredDoctors = new ArrayList<DoctorUserEntity>();
         return doctors;
     }
 
+    private void calculateRatings() {
+        float sum = 0;
+        int count = 0;
+        for (DoctorUserEntity doctorUserEntity : doctors) {
+            for (AppointmentEntity appointmentEntity : doctorUserEntity.getAppointmentList()) {
+                if (appointmentEntity.getRate() != 0) {
+                    sum += appointmentEntity.getRate();
+                    count++;
+                }
+            }
+            if (count == 0) {
+                doctorUserEntity.setRating(0);
+            } else {
+                doctorUserEntity.setRating(sum/count);
+            }
+            sum = count = 0;
+        }
+    }
```

Comment:
This method should not be in here. You should instead write a getRating() method with these codes inside DoctorUserEntity to get the rating of a doctor.